### PR TITLE
Add PAM Metric for Reference-Free Audio Quality

### PR DESCRIPTION
**Pull request by DareenHarthi**
This Pull Request adds support for the PAM (Prompting Audio-Language Models) metric, a reference-free metric introduced in the paper: [PAM: Prompting Audio-Language Models for Audio Quality Assessment](https://github.com/soham97/PAM/).

Main Changes:

- utterance_metrics Folder: Added pam.y to compute the PAM metric for given audio samples.
- versa/bin Folder: Added clap.py to extract CLAP embeddings, which are necessary for computing PAM.
- PAM Configuration: Added pam.yaml in egs/separate_metrics for configuring PAM-related settings.
- Updates to versa/scorer_shared.py: Integrated the PAM metric into the load_score_modules() and use_score_modules() functions to enable its usage.
- Updated the list_scoring() function at line 555: Modified the condition elif io == "soundfile" to elif io == "soundfile" or io == "dir" to handle directory input.

Requested Feature

- For PAM, there is an option to: Use all segments of the audio, or use only the first segment. Since CLAP only accepts audio segments of 7 seconds or less, if the audio is longer, it is split into smaller segments, and the average scores are computed. We need to add argument to choose either to use the first segment or all the segments.

### CODE REVIEW

File: test_result [hunk 1]
```diff
@@ -0,0 +1 @@
+{'key': 'test.wav', 'pam_score': 0.01386283989995718}
```
Comment (ftshijt) [inline test_result:1]: We add some specific test scripts for individual models at test/test_{metric}.py. You may consider refer to that to add the test function~

File: README.md [hunk 2]
```diff
@@ -135,6 +135,7 @@ We include [ ] and [x] to mark if the metirc is auto-installed in versa.
 | 36 | Speech enhancement-based CI-SDR [x] | se_snr | se_ci_sdr | [ESPnet](https://github.com/espnet/espnet.git) | |
 | 37 | Speech enhancement-based SAR [x] | se_snr | se_sar | [ESPnet](https://github.com/espnet/espnet.git) | |
 | 38 | Speech enhancement-based SDR [x] | se_snr | se_sdr | [ESPnet](https://github.com/espnet/espnet.git) | |
+| 39 | Audio Quality | pam | pam | [PAM](https://github.com/soham97/PAM/tree/main) | [Paper](https://arxiv.org/pdf/2402.00282)|
```
Comment (ftshijt) [inline README.md:138]: We may consider using "Prompting Audio-Language Models (PAM)-based Audio Quality Evaluation" for the second column (metric name)

**Review by ftshijt — COMMENTED** [review]
Many thanks! It would be great to realize PAM in the setup. I add a few interface-related suggestions, which could be better to fix before our merge.

**Review by DareenHarthi — COMMENTED** [review]
I've addressed all comments. Let me know if further changes are needed!

**Review by ftshijt — COMMENTED** [review]
Many thanks! Minor comments as below:

File: versa/bin/clap.py [hunk 3]
```diff
@@ -0,0 +1,1090 @@
+import numpy as np
+import math
+import random
+import torch
+import torch.nn.functional as F
+from torch import nn
+from transformers import AutoModel
+import torch.utils.checkpoint as checkpoint
+from torchlibrosa.stft import Spectrogram, LogmelFilterBank
```
[the remark this answers is not in the document]
Comment (DareenHarthi) [reply versa/bin/clap.py:9]: I need only torchlibrosa, transformers is already in the setup file.

**Comment by ftshijt** [comment]
Many thanks for your contribution!
